hkmc2/shared/src/main/scala/hkmc2/semantics/Term.scala

```diff
@@ -72,8 +76,8 @@ enum Term extends Statement:
   case Missing // Placeholder terms that were not elaborated due to the "lightweight" elaboration mode `Mode.Light`
   case Lit(lit: Literal)
   case Builtin(id: Tree.Ident, nme: Str)
-  case Ref(sym: Symbol)(val tree: Tree.Ident, val refNum: Int) extends Term with Resolvable
-  case App(lhs: Term, rhs: Term)(val tree: Tree.App, var resultSym: Opt[FieldSymbol], val resSym: FlowSymbol) extends Term with Resolvable
+  case Ref(var sym: Symbol)(val tree: Tree.Ident, val refNum: Int) extends Term with Resolvable
```

Comment:
Do we ever need to _change_ the symbol of a `Ref`? 🤔

Reply:
Hmm... no then it would be lowered to something else... Probably we have to add a new field for Ref?

Reply:
Yeah, don't change the symbol that a reference refers to, that's crazy!